src/twisted/conch/client/default.py

```diff
@@ -130,6 +130,31 @@ def isInKnownHosts(host, pubKey, options):
 
 
 
+def getHostKeyAlgorithms(host, options):
+    """
+    Look in known_hosts for a key corresponding to C{host}.
+    This can be used to change the order of supported key types
+    in the KEXINIT packet.
+
+    @type host: L{str}
+    @param host: the host to check in known_hosts
+    @type options: L{twisted.conch.client.options.ConchOptions}
```

Comment:
Why does this take a `ConchOptions` rather than a `KnownHostsFile`?  It seems that this is duplicating the logic involved in determining the proper location of the `known_hosts` file.